subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -1143,6 +1145,9 @@ int ztls_connect(int sock, const struct sockaddr *addr, socklen_t addrlen)
 				goto error;
 			}
 
+			/* Do not use any socket flags during the handshake. */
+			context->tls->flags = 0;
```

Comment:
is it ok override flags here and restore after handshake ?

Reply:
The flags are set explicitly during each `send`/`recv' call. As for `connect`/`accept` there is no way to pass flags to a socket, but we still use it during the handshake, therefore we should clear any leftover flags. Still, at this point it's more like an extra precaution, as TLS socket is currently not reusable and should be closed after connection is closed.